scripts/core/superdesk-api.d.ts

```diff
@@ -947,29 +954,37 @@ declare module 'superdesk-api' {
         direction: 'ascending' | 'descending';
     }
 
+    export interface ICrudManagerResponse<T> {
```

Comment:
not sure about the naming, what does crud manager response mean

Reply:
`CRUD` stands for "create, read, update, delete" https://en.wikipedia.org/wiki/Create,_read,_update_and_delete

In this case, crud manager is an interface for storage of lists. I initially created it for generic list view(used in annotations library, internal destinations page), which stores data via HTTP API calls. This time, I wanted to reuse the same UI from generic list view, but the data I wanted to manage wasn't backed by HTTP API collections, but was a simple array. To make it work, I created an implementation of crud manager interface that works with an array.

Reply:
I know the crud, I was rather concerned about the manager response because that crud manager is not responding to anything, is it?

Reply:
Correct, crud manager itself is not responding to anything. Its methods do - `read`, `sort`, `refresh`, `removeFilter`, `goToPage`. The interface defines common data format that these methods respond with. I don't mind the name, feel free to suggest another which is clearer to you :)

![2022-02-16_09-40](https://user-images.githubusercontent.com/2076953/154227119-3542b697-e6aa-49a1-a17d-d4fd6b738dd5.png)

Reply:
`ICrudManagerState` has more fields than `ICrudManagerResponse` which wouldn't be available in crud manager methods. I think the interfaces themselves are fine and I doubt that it would work to simplify them easily. What we could do is rename if it would improve clarity.

Reply:
ok, would probably call it ICrudManagerData then just to avoid calling it response when it's not